src/main/java/org/folio/services/invoice/PoLinePaymentStatusUpdateService.java

```diff
@@ -109,6 +120,81 @@ public Future<Void> updatePoLinePaymentStatusToCancelInvoice(Invoice invoiceFrom
       });
   }
 
+  /**
+   * Updates payment status of the associated PO Lines (when poLinePaymentStatus is not used)
+   *
+   * @param invoiceLines the invoice lines to be paid
+   * @return CompletableFuture that indicates when transition is completed
+   */
+  private Future<Void> payPoLines(List<InvoiceLine> invoiceLines, RequestContext requestContext) {
+    Map<String, List<InvoiceLine>> poLineIdInvoiceLinesMap = groupInvoiceLinesByPoLineId(invoiceLines);
+    return fetchPoLines(poLineIdInvoiceLinesMap, requestContext)
+      .map(this::updatePoLinesPaymentStatus)
+      .compose(poLines -> orderLineService.updateCompositePoLines(poLines, requestContext));
+  }
+
+  private Map<String, List<InvoiceLine>>  groupInvoiceLinesByPoLineId(List<InvoiceLine> invoiceLines) {
+    if (CollectionUtils.isEmpty(invoiceLines)) {
+      return Collections.emptyMap();
+    }
+    return invoiceLines
+      .stream()
+      .filter(invoiceLine -> StringUtils.isNotEmpty(invoiceLine.getPoLineId()))
+      .collect(Collectors.groupingBy(InvoiceLine::getPoLineId));
```

Comment:
Can be simplified aswell:
```suggestion
    return StreamEx.of(invoiceLines)
      .filter(invoiceLine -> StringUtils.isNotEmpty(invoiceLine.getPoLineId()))
      .groupingBy(InvoiceLine::getPoLineId);
```

Reply:
Same as above.